### PR TITLE
Add background/overlay imagery layer info button

### PR DESCRIPTION
**Pull request by zedlabs**
creates #1358

### CODE REVIEW

File: src/main/java/de/blau/android/dialogs/ImageryListAdapter.java [hunk 1]
```diff
@@ -59,11 +66,21 @@ public ImageryListAdapter(@NonNull String[] ids, String currentId, @Nullable boo
         }
     };
 
+    interface OnInfoClickListener{
+        void onItemClick(String id);
+    }
+
+    public void addItemClickListener(OnInfoClickListener listener) {
```
Comment (simonpoole) [inline src/main/java/de/blau/android/dialogs/ImageryListAdapter.java:73]: As above: consistent naming, javadoc and annotations.

File: src/main/java/de/blau/android/dialogs/Layers.java [hunk 2]
```diff
@@ -701,6 +702,7 @@ private AlertDialog buildImagerySelectDialog(@Nullable final TableRow row, @Null
 
         final ImageryListAdapter adapter = new ImageryListAdapter(ids, currentId, isOverlay, buttonLayoutParams,
                 new LayerOnCheckedChangeListener(activity, dialog, row, layer, ids));
+        adapter.addItemClickListener(this);
```
Comment (simonpoole) [inline src/main/java/de/blau/android/dialogs/Layers.java:705]: While a bit a matter of taste, I would have simply provided the listener as a lambda here, so something like

```
      adapter.addItemClickListener((String id) -> {
            TileLayerSource l = TileLayerSource.get(getContext(), id, true);
            if (l != null) {
                LayerInfo f = new ImageryLayerInfo();
                f.setShowsDialog(true);
                Bundle args = new Bundle();
                args.putSerializable(ImageryLayerInfo.LAYER_KEY, l);
                f.setArguments(args);
                LayerInfo.showDialog(getActivity(), f);
            }
        });
```

File: src/main/java/de/blau/android/dialogs/Layers.java [hunk 3]
```diff
@@ -719,6 +721,22 @@ private AlertDialog buildImagerySelectDialog(@Nullable final TableRow row, @Null
...
+        if (layer != null) {
+            LayerInfo f = new ImageryLayerInfo();
+            f.setShowsDialog(true);
+            Bundle args = new Bundle();
+            args.putSerializable(ImageryLayerInfo.LAYER_KEY, layer);
+            f.setArguments(args);
+            LayerInfo.showDialog(getActivity(), f);
+        }
+    }
+
```
Comment (simonpoole) [inline src/main/java/de/blau/android/dialogs/Layers.java:739]: See above. as we might want to rip the whole imagery selection dialog out and put it in a separate class one day, I would tend to keep everything together.

File: src/main/res/layout/imagery_layer_list_item.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"/>
+    <ImageButton
+        android:id="@+id/listItemInfo"
+        android:layout_height="wrap_content"
+        android:layout_width="wrap_content"
+        android:layout_alignParentRight="true"
+        android:layout_centerInParent="true"
+        android:background="@android:color/transparent"
+        android:src="?attr/menu_information"
```
Comment (simonpoole) [inline src/main/res/layout/imagery_layer_list_item.xml:18]: I suspect ?attr/information_small would look better here (it's a slightly smaller version of the same icon). 

The other thing is that when I tested in an emulator there was some overlap between the text and the (i) in some cases as  the text wasn't wrapping early enough, could you check that?

**Review by simonpoole — CHANGES_REQUESTED** [review]
Looks good, some minor nitpicks are listed inline below.

**Comment by zedlabs** [comment]
made the changes but the icon seems to small
Screenshot (163)

**Comment by simonpoole** [comment]
Hmm, do you have font scaling turned on? For me (standard emulator settings), it looks like 

Screenshot_1615996931

While one could still argue that the button is too small (at least for me it doesn't seem so in practical terms on a real device, and I have not so good eyesight), the proportions would seem to fit better.

**Comment by zedlabs** [comment]
Yeah, I forgot to turn scaling off.
It looks fine on the default font size

**Review by simonpoole — APPROVED** [review]

**Comment by simonpoole** [comment]
Thanks for the contribution. 

We're just before a release, which will then be on its own branch.  Potentially I'll ask you to rebase immediately after that has happend (the only expected changes are translations and documentation) so that I can cleanly merge this.

**Comment by simonpoole** [comment]
@zedlabs can you rebase this on master? Thanks.

**Comment by zedlabs** [comment]
done @simonpoole

**Pull request merged**